src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -74,7 +77,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
      */
     String w3Link = getLink(W3_CSE_ID, query);
     if (w3Link != null) {
-      allLinks.add(w3Link);
+      W3SchoolClient client = new W3SchoolClient();
```

Comment:
So I notice that these clients have zero internal state, which makes sense. This makes search() feel like something that should be a static method on the W3SchoolClient (and friends) class.

Alternatively I also like Joan's proposal of making an interface for these clients. My only addition to it is to not instantiate a client on every request - just make a `List<Client>` in the constructor.

Reply:
done by adding interface and list